### PR TITLE
PM-295: Duplicate Share Bug Fix

### PR DESCRIPTION
**Pull request by andre-meyer**
Also removes `market.shares` from market object - selector handles this now

### CODE REVIEW

File: src/api/schema.js [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
 import { hexWithPrefix } from 'utils/helpers'
 
+const normalizeShareObject = share => ({
+  event: share.outcomeToken.event,
+  ...share,
+  outcomeToken: mapValues(share.outcomeToken, hexWithPrefix),
+})
+
+const shareMerge = (shareA, shareB) => {
+  console.warn('merging', shareA, shareB)
```
Comment (mmvsha) [inline src/api/schema.js]: Do we need this in prod?

File: src/components/Dashboard/index.js [hunk 2]
```diff
@@ -201,9 +201,9 @@ class Dashboard extends Component {
           </div>
         )
       }
+    }).filter(val => !!val)
 
-      return <div />
-    })
+    return myShares.length ? myShares : <div>You aren't holding any share.</div>
```
Comment (mmvsha) [inline src/components/Dashboard/index.js]: I remember you asked me to fix this 😄

Comment (andre-meyer) [reply src/components/Dashboard/index.js]: Too late! 😁

File: src/api/schema.js (hunk 1 above)
Comment (andre-meyer) [reply src/api/schema.js]: Yes, I want to see in sentry when we had to merge. Should never happen but when it happens I want to see it in the log

File: src/api/schema.js [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+  return {
+    ...shareA,
+    ...shareB,
+    balance: Decimal(shareA.balance).add(Decimal(shareB.balance)).toString(),
```
Comment (mmvsha) [inline src/api/schema.js]: Why do you make `shareB.balance` an instance of Decimal? Decimal accepts strings as arguments

File: src/components/Dashboard/index.js (hunk 2 above)
Comment (andre-meyer) [reply src/components/Dashboard/index.js]: Couldn't be more dirty but we need a complete new selector for this to implement it cleanly - let's refactor after launch

**Pull request merged**